AstSemantics.md

```diff
@@ -202,11 +202,15 @@ Out of bounds accesses trap.
 
 In the MVP, linear memory can be resized by a `grow_memory` operator. The
 operand to this operator is in units of the WebAssembly page size,
-which is 64KiB on all engines (though large page support may be added in 
+which is defined to be 64KiB (though large page support may be added in 
 the [future](FutureFeatures.md#large-page-support)).
 
  * `grow_memory` : grow linear memory by a given unsigned delta of pages.
-    Return the previous memory size in bytes.
+    Return the previous memory size in units of pages or -1 on failure.
+
+`grow_memory` must fail when attempting to grow past the maximum declared
+size, if one is specified in the [memory section](Module.md#linear-memory-section).
+`grow_memory` may also fail if an underlying system allocation fails.
```

Comment:
There's some subtley here because:
* if a maximum *is* set, we can still fail before it is reached (both if our initial reservation wasn't able to grab maximum or, if we're using memory protection tricks, if the `mprotect` to enable the memory fails (rare as that may be))
* if a maximum *is not* set, on 64-bit an engine may still choose to reserve 4GiB (b/c bounds check elimination and vaddr space is no longer a precious resource.

I pushed an update that I think addresses your concern while respecting these subtleties.  PTAL